### ISSUE TITLE
Genres are not returned for albums

### ISSUE THREAD

**Issue by RocketCity** [spotify/web-api#157]
The list of genres for an album is always empty: confirmed in the [Web API Console for "Get an Album"](https://developer.spotify.com/web-api/console/get-album/) -- I tried it in code for hundreds of albums and none return a genre.

**Comment by holmberd** [comment]
*Get an artist* endpoint seems to be the only endpoint that returns genres for some artists.

**Comment by holmberd** [comment]
The only good API that was able to deliver good genre data was Echo Nest. But after Spotify acquired Echo Nest in 2014, most of there support stopped and API rate-limits went from 120 to 20. It is currently a ghost company, as there twitter, fb, forum and blog are all dead.

**Comment by aljondo** [comment]
This is a problem for me too, is it going to be fixed soon?

**Comment by AltroozTodd** [comment]
+1

**Comment by gaurav183** [comment]
Any updates? +1

**Comment by holmberd** [comment]
>  We don't have weights for artists genres because that information isn't available to us, and genres are so subjective that the value of such a weighting would be limited.

 Why is this considered a bug? When Spotify aquired EchoNest, which had endpoints for all the data I needed, they should have gained access to all the genres available from that service.

**Comment by jpmurray** [comment]
I know it's bad practice, but I'll just add me voice to this, at least to get notified of any updates on the matter.

Now, if the genres aren't supposed to be exposed on the album object, you might as well remove it. (I'd rather them to be there though!) Seeing that someone said in #31 that we _should_ be able to get them by album, it really seems like a bug. And a long standing one at that :(

That being said, I'm adding also my voice to @holmberd on this one. If I remember correctly, EchoNest also had genre data to the track level, it's weird we can't access the data anymore. If I may add, in the linked issue above, we're said that Spotify doesn't _expose_ the track level genre data, but it doesn't seems it's not there. This kind of information in invaluable to some, and I'd honestly be willing to pay to access an API that gives me track genres (since it seems no current api are willing to do so!).

**Comment by goldensunliu** [comment]
@thelinmichael since you no longer work at Spotify. Could you help point us the right person in terms of the most up to date information regarding the web api endpoints?

**Comment by davidgibsonp** [comment]
+1

**Comment by michelle-chiang** [comment]
+1

**Comment by rosem1696** [comment]
+1

**Comment by fredericklu** [comment]
+1

**Comment by ssemenova** [comment]
+1

**Comment by sureshHARDIYA** [comment]
+ 1

**Comment by jul1an** [comment]
+1

**Comment by grahamalama** [comment]
+1

**Comment by Rawnly** [comment]
+1

**Comment by travisR004** [comment]
+1

**Comment by jpmurray** [comment]
PEOPLE. Stop saying `+1`, it's only spamming previous commenters and it's, obviously at this point, not going to motivate anyone from this repo to act on it! Use reactions on original post!

**Comment by arirawr** [comment]
Hey all,

As per our [new developer support process](https://github.com/spotify/web-api#creating-an-issue), any issues that have no plans to be considered in the next 6 months will be closed as wontfix.

We do not have any plans to implement this in the next 6 months and thus will close as wontfix.

We encourage you and other members of the community to continue to share use cases and your support for this feature, as it helps us prioritize features going forward. You will be able to comment on this issue even after it is closed. We will also post any updates here relating to this suggestion when they happen.

**Comment by jpmurray** [comment]
@arirawr it is, at minimum, possible to get the reasoning? I mean, if it's not possible within 6 month, is it because at the start you don't _have_ the information, or is it because you're _not planning to share_ the information?

This gives us a good idea if what _we_ want to do is actually possible or not.

**Comment by shillshocked** [comment]
Bump!

**Comment by Anoesj** [comment]
Almost one year later.. Bump!

**Comment by buckmaxwell** [comment]
Posting here because the more appropriate #263 was closed claiming it was a duplicate.

The GET /artists/{id} endpoint and GET /artists?ids={commaSeparatedListOfIds} endpoints are flawed.

Out of a sample of 2,317 artists only 1,785 had genres associated. Out of 1166 artists with popularity over 50, 39 had no genres returned. More than one of these artists had popularity > 70.

What gives? it seems that Spotify is aware of this information.

**Comment by jpmurray** [comment]
@buckmaxwell I'm starting to feel like they don't want genre tu appear on purpose. Since Spotify are creating their own playlist, probably based on genre, they would not want any external tool to do the same, and maybe better?

I get the policy that got this closed, but the lack on _reasons_ still bugs me to date.

**Comment by buckmaxwell** [comment]
@jpmurray it's possible. Overall their API is so generous though, esp. w the non-commercial license. I wonder if they've actually gone away from using 'genre' internally and are focused more on the audio features and analysis? or info about related artists. In that case they might not bother to keep the old data they got from echo nest up to date. Haven't checked if there is a bias towards newer songs not having genres though.

**Comment by jpmurray** [comment]
@buckmaxwell Honestly, if they decided to _let go_ of old data, that'd be more of a reason to release said info!

**Comment by chuck-stein** [comment]
I would also like to say this is still an issue for me and is a complete blocker for a personal-use app I started making whose purpose is to recommend albums from your own library based on genres you're currently in the mood for. I decided to go through with the app after initially researching that the API's albums endpoint does indeed return genre data according to the reference documentation, so it is disheartening to put work into my app only to find out this is not actually the case. Would greatly appreciate any update from the Spotify team, as it has been more than 6 months!
